src/state/order_schema/order_packet.rs

```diff
@@ -739,7 +778,9 @@ fn test_decode_order_packet() {
         };
         let bytes = packet.try_to_vec().unwrap();
         let decoded_normal = decode_order_packet(&bytes).unwrap();
-        let decoded_inferred = decode_order_packet(&bytes[..bytes.len() - 2]).unwrap();
+        let num_stripped_bytes = if rng.gen::<f64>() > 0.5 { 1 } else { 3 };
+        let decoded_inferred =
+            decode_order_packet(&bytes[..bytes.len() - num_stripped_bytes]).unwrap();
```

Comment:
we should test both cases deterministically here and below

Reply:
oh i see what you mean. will do